source/content.js

```diff
@@ -44,6 +45,17 @@ function onNewTweets(cb) {
 	observeEl('#stream-items-id', cb);
 }
 
+function onSingleTweetOpen(cb) {
+	observeEl('body', mutations => {
+		for (let i = 0; i < mutations.length; i++) {
```

Comment:
So it's kinda troublesome to detected when a tweet is open: `onpopstate` is a no go, since Twitter uses `pushState` to "navigate" to a single tweet and I didn't want to monkey patch the `history.pushState` native method.

Following @sindresorhus recommendation, I've added an observer to the DOM that checks if the class `overlay-enabled` was added to the body. Unfortunately, this was not enough, as the tweet will be loaded just AFTER the class is appended.

So I added a second observer that is added to `#permalink-overlay` and that is the observer that will run the callback.

I wasn't sure If this needed to be split into two functions, so I just used an arrow function, but let me know how you feel about this.

Reply:
Why can't we just monkey-patch `history.pushState`? That would be much more efficient and we could use it for other route changes too. We can just tap it without modifying the behavior. I can't think of any downside. Can you?

```js
const pushState = history.pushState;
history.pushState = (...args) {
	Reflect.apply(pushState, history, args);
	cb();
};
```

Reply:
I just didn't want to risk the user having another extension that also taps into `pushState` and things breaking, but maybe I'm just being paranoid. I'll update the PR with the patch and we can look into this in the future if it's ever an issue.

Reply:
@sindresorhus so, turns out tapping into `history.pushState` does not work. Seems link twitter binds a new function from `history.pushState` before the extension is run, so our tap never gets executed. Did you manage to make it work? Not even running right from the console I was able to get a satisfying result.

I did find a `window` scoped object (`window.ttft`, to be precise) we could watch to detect the navigation but nothing that I'm too excited about.

Reply:
Yeah, I forgot that content scripts only has access to the page dom, not the `window` object. So monkey patching won't work. We can go with your way then.

Reply:
> for (let i = 0; i < mutations.length; i++) {

Use a `for-of` loop here.